sycl/include/sycl/ext/oneapi/experimental/annotated_arg/annotated_arg.hpp

```diff
@@ -146,13 +156,17 @@ __SYCL_TYPE(annotated_arg) annotated_arg<T *, detail::properties_t<Props...>> {
                          const PropertyListV &proplist) noexcept
       : obj(other.obj) {
     (void)proplist;
-    static_assert(std::is_convertible<T2, T *>::value,
+    static constexpr bool is_input_convertible =
+        std::is_convertible<T2, T *>::value;
+    static_assert(is_input_convertible,
                   "The underlying data type of the input annotated_arg is not "
```

Comment:
This is also vague "underlying data type is not compatible" - not compatible how? How is the user supposed to correct their code with that? Yug, please take care of this in your next PR where you rewrite the other error messages.